File: typescript-sdk/src/policies.ts
/**
 * Default policy configurations for Anchor agents.
 */
export interface DefaultPolicyPackConfig {
    allowedDomains?: string[] | null;
    maxQuerySize?: number | null;
    requireApprovalFor?: string[] | null;
    blockPii?: boolean;
    blockSecrets?: boolean;
}

const DEFAULT_POLICY_CONFIG = {
    maxQuerySize: 1000,
    requireApprovalFor: ['delete', 'update', 'export'],
    blockPii: true,
    blockSecrets: true,
} as const;

/**
 * Production ready default policies for agent config.
 *
 * @example
 * ```typescript
 * import { DefaultPolicyPack } from 'anchorai';
 *
 * // Use default policies
 * await anchor.config.update(agent.id, new DefaultPolicyPack().getConfig());
 *
 * // Customization (update default values / disable certain policies)
 * await anchor.config.update(agent.id, new DefaultPolicyPack({
 *   blockSecrets: false,
 *   maxQuerySize: null
 * }).getConfig());
 * ```
 */
export class DefaultPolicyPack {
    private config: DefaultPolicyPackConfig;

    constructor(config: DefaultPolicyPackConfig = {}) {
        this.config = config;
    }

    /**
     * Returns an object with configured policies to be used in `config`
     */
    getConfig(): { policies: Record<string, any> } {
        const policies: Record<string, any> = {};
        const { config } = this;

        if (Array.isArray(config.allowedDomains)) {
            policies.allowed_domains = config.allowedDomains;
        }

        const maxQuerySize =
            config.maxQuerySize === undefined
                ? DEFAULT_POLICY_CONFIG.maxQuerySize
                : config.maxQuerySize;

        if (maxQuerySize !== null) {
            policies.max_query_size = maxQuerySize;
        }

        const requireApprovalFor =
            config.requireApprovalFor === undefined
                ? DEFAULT_POLICY_CONFIG.requireApprovalFor
                : config.requireApprovalFor;

        if (requireApprovalFor !== null) {
            policies.require_approval_for = requireApprovalFor;
        }

        policies.block_pii = config.blockPii ?? DEFAULT_POLICY_CONFIG.blockPii;
        policies.block_secrets = config.blockSecrets ?? DEFAULT_POLICY_CONFIG.blockSecrets;

        return { policies };
    }
}

File: python-sdk/src/anchor/policies.py
"""Default policy configurations for Anchor agents."""

from dataclasses import dataclass, field
from typing import Any, Optional


@dataclass
class DefaultPolicyPack:
    """
    Production ready default policies for agent config.

    Usage:
        from anchor import DefaultPolicyPack

        # Apply defaults
        anchor.config.update(agent.id, DefaultPolicyPack().get_config())

        # Customize defaults
        pack = DefaultPolicyPack(
            block_secrets=False,
            max_query_size=None  # Disable specific policy
        )
        anchor.config.update(agent.id, pack.get_config())
    """

    allowed_domains: Optional[list[str]] = None
    max_query_size: Optional[int] = 1000
    require_approval_for: Optional[list[str]] = field(
        default_factory=lambda: ["delete", "update", "export"]
    )
    block_pii: bool = True
    block_secrets: bool = True

    def get_config(self) -> dict[str, Any]:
        """
        Returns dict with configured policies to be used in `config`
        """
        policies: dict[str, Any] = {}

        if self.allowed_domains is not None:
            policies["allowed_domains"] = self.allowed_domains

        if self.max_query_size is not None:
            policies["max_query_size"] = self.max_query_size

        if self.require_approval_for is not None:
            policies["require_approval_for"] = self.require_approval_for

        policies["block_pii"] = self.block_pii
        policies["block_secrets"] = self.block_secrets

        return {"policies": policies}
